gen/vsup_zimbra

```diff
@@ -54,32 +53,44 @@ my $users;  # $users->{$uco}->{ATTR} = $attrValue;
 # AGGREGATE DATA
 #
 # FOR EACH USER
-foreach my $rData ($data->getChildElements) {
+foreach my $resourceId ($data->getResourceIds()) {
 
-	my %resourceAttributes = attributesToHash $rData->getAttributes;
-	my $relationType = $resourceAttributes{$A_R_RELATION_TYPE};
+	my $relationType = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_R_RELATION_TYPE);
 
 	# Users from Resource must be in a relation
 	unless ($relationType) {
 		next;
 	}
 
-	my @membersData = $rData->getChildElements;
+	foreach my $memberId ($data->getMemberIdsForResource(resource => $resourceId)) {
 
-	foreach my $member (@membersData) {
-
-		my %uAttributes = attributesToHash $member->getAttributes;
+		my $login  = $data->getUserFacilityAttributeValue( member => $memberId, attrName => $A_LOGIN);
 
 		# SKIP MEMBERS WHICH SUPPOSED TO BE SYSTEM WIDE ACCOUNTS => IGNORED BY PERUN
-		if (exists $ignoredAccounts{$uAttributes{$A_LOGIN}}) {
+		if (exists $ignoredAccounts{$login}) {
 			next;
 		}
 
-		my $uco = $uAttributes{$A_UCO};
-		$users->{$uco}->{$A_LOGIN} = $uAttributes{$A_LOGIN};
-		$users->{$uco}->{'EMAIL'} = ($uAttributes{$A_EMAIL_VSUP} || $uAttributes{$A_LOGIN} . '@vsup.cz');
+		my $uco  = $data->getUserAttributeValue(member => $memberId, attrName => $A_UCO);
+		my $vsupMail  = $data->getUserAttributeValue(member => $memberId, attrName => $A_EMAIL_VSUP);
+		my $userId  = $data->getUserAttributeValue(member => $memberId, attrName => $A_USER_ID);
+		my $artisticFirstName = $data->getUserAttributeValue(member => $memberId, attrName => $A_ARTISTIC_FIRST_NAME);
+		my $uFirstName = $data->getUserAttributeValue(member => $memberId, attrName => $A_FIRST_NAME);
+		my $artisticLastName = $data->getUserAttributeValue(member => $memberId, attrName => $A_ARTISTIC_LAST_NAME);
+		my $uLastName = $data->getUserAttributeValue(member => $memberId, attrName => $A_LAST_NAME);
+		my $uDisplayName = $data->getUserAttributeValue(member => $memberId, attrName => $A_ZIMBRA_DISPLAY_NAME);
+		my $blacklisted = $data->getUserFacilityAttributeValue(member => $memberId, attrName => $A_BLACKLISTED);
+		my $expirationKOS = $data->getUserAttributeValue(member => $memberId, attrName => $A_EXPIRATION_KOS);
+		my $expirationDC2 = $data->getUserAttributeValue(member => $memberId, attrName => $A_EXPIRATION_DC2);
+		my $expirationManual = $data->getUserAttributeValue(member => $memberId, attrName => $A_EXPIRATION_MANUAL);
+		my $vsupMailAlias = $data->getUserAttributeValue(member => $memberId, attrName => $A_EMAIL_VSUP_ALIAS);
+		my $vsupMailAliases = $data->getUserAttributeValue(member => $memberId, attrName => $A_EMAIL_VSUP_ALIASES);
```

Comment:
Following attributes are manually retrieved from RPC and they are not present in hashed data structure.

Reply:
Fixed